sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -3878,6 +3880,8 @@ class Dataset[T] private[sql](
    */
   def persist(newLevel: StorageLevel): this.type = {
     sparkSession.sharedState.cacheManager.cacheQuery(this, None, newLevel)
+    queryExecution = new QueryExecution(
```

Comment:
1. Persisting in `cacheManager` is global - it affects all `Dataset` instances.
2. `unpersist` on any of `Dataset` instances would require updating `queryExecution` again on all instances with the same `logicalPlan`

Reply:
So when we judge whether it's necessary to renew a `queryExecutionPersisted` instance, we always need transform down the whole queryExecution's logical plan tree and lookup it's children in the `cacheManager`.